gdal/frmts/sentinel2/sentinel2dataset.cpp

```diff
@@ -809,6 +847,11 @@ static bool SENTINEL2GetGranuleList(CPLXMLNode* psMainMTD,
     }
 #endif
 
+    const bool isMSI2Ap = EQUAL(CPLGetXMLValue(psProductInfo, "PRODUCT_TYPE", ""),
```

Comment:
No need for the ternary ? : operator. EQUAL() returns a bool

Reply:
Code has been changed accordingly.